cpp/include/cudf/strings/string_view.inl

```diff
@@ -21,15 +21,19 @@ namespace
 {
 using BYTE = uint8_t;
 
-/**---------------------------------------------------------------------------*
+// number of characters in a string computed on-demand
+// the _nchars member is initialized to this value as a place-holder
+static constexpr cudf::size_type UNKNOWN_STRING_LENGTH{-1};
```

Comment:
I see no reason for this to be static. The only reason for static here would be internal linkage, which is satisfied by anonymous namespaces in C++.

```suggestion
constexpr cudf::size_type UNKNOWN_STRING_LENGTH{-1};
```